============================= test session starts ==============================
platform linux -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/clapp_stanford_edu/Src/sep-python
collected 7 items / 6 deselected / 1 selected

test_sepfile.py F                                                        [100%]

=================================== FAILURES ===================================
____________________________ test_simpleArgs_1Line _____________________________

    def test_simpleArgs_1Line():
        f=open("/tmp/junk.H","w")
        f.write("\n n1=10 o1=1. d1=2. label='test' in='/tmp/junk.H@'\n")
        f.close()
>       fl=sepPython.sepIO.sFile(path="/tmp/junk.H")

test_sepfile.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py:435: in __init__
    super().__init__(**kw)
/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py:196: in __init__
    self._params=self.buildParamsFromPath(kw["path"],**kw)
/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py:221: in buildParamsFromPath
    pars=self.getHistoryDict(fle)
/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py:471: in getHistoryDict
    pars=databaseFromStr(self._history,pars)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

strIn = b"\n n1=10 o1=1. d1=2. label='test' in='/tmp/junk.H@'\n", dataB = {}

    def databaseFromStr(strIn:str,dataB:dict):
>     lines=strIn.split("\n")
E     TypeError: a bytes-like object is required, not 'str'

/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py:67: TypeError
=========================== short test summary info ============================
FAILED test_sepfile.py::test_simpleArgs_1Line - TypeError: a bytes-like objec...
======================= 1 failed, 6 deselected in 1.17s ========================
Exception ignored in: <function regFile.__del__ at 0x7f207aa4e950>
Traceback (most recent call last):
  File "/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/ioBase.py", line 209, in __del__
  File "/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py", line 389, in close
  File "/home/spack/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.5.0/python-3.10.8-tye3styllri3xiql6psptyz7uujbz4hn/lib/python3.10/site-packages/sepPython/sepIO.py", line 546, in writeDescription
NameError: name 'open' is not defined
